.github/workflows/security-scan.yml

```diff
@@ -0,0 +1,92 @@
+name: Security Scan
+
+on:
+  push:
+    branches:
+      - main
+      - release/**
+  pull_request:
+    branches:
+      - main
+      - release/**
```

Comment:
Security: for my own understanding, is the visibility of results on PRs automagical via SARIF meta? My understanding is we can get warnings in PRs for net-new vulns by enabling scans on PR events, but I'm not 100% certain I've done enough here for that to be the case.

Reply:
These results _should_ be automagical, because they're uploaded here:

https://github.com/hashicorp/consul/blob/527a7ce45d35927730e7f7e56f78d4aab7ac1e54/.github/workflows/security-scan.yml#L89-L92

Any _net new_ results (not already on `main`, for example) will be shown in-line of the PR, in the security tab (for the PR), or in the "checks" tab of a PR.